EmployeeLib/Abstract/IHasManager.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace EmployeeLib.Abstract
+{
+    public interface IHasManager<T> where T : IBaseEmployee
```

Comment:
The generic type T should not be of type IBaseEmployee, if you add a developer as a manager it will work. which is wrong. this should be of type manager